lib/ansible/modules/cloud/azure/azure_rm_image.py

```diff
@@ -0,0 +1,291 @@
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = '''
+---
+author: Emma Laurijssens van Engelenhoven
+description:
+  - Capture an Azure Virtual Machine image for the deployment of other VMs in Azure
+     The VM should be generalized using sysprep. After this command runs, the Virtual Machine will be unusable.
+     Depends on pip azure module 2.0.0 or above and azure-mgmt-compute 2.1.0 and above.
+module: azure_rm_image
+options:
+  resource_group_name:
+    description:
+      - The resource group name to use or create to host the deployed template
+    required: true
+  name:
+    description:
+      - The name of the image being created.
+    required: True
+  vm_name:
+    description:
+      - The name of the VM within the resource group to capture. Required when state=present.
+    required: false
+  location:
+    description:
+      - The location where the image should be stored. Should be within the resource group.
+    default: The location of the resource group
+    required: false
+  state:
+    description:
+      - The desired state of the image
+    default: present
+    choices: [present, absent]
+    required: false
+  tags:
+    description:
+        - Tags to assign to the image.
+    required: false
+
+short_description: Capture Azure Virtual Machine Images
+version_added: "2.5"
+
+'''
+
+EXAMPLES = '''
+- name: Capture image
+  hosts: 127.0.0.1
+  connection: local
+
+  tasks:
+    - name: Capture image
+      azure_rm_image:
+        subscription_id: "{{ subscription_id }}"
+        resource_group_name: "{{ resource_group_name }}"
+        client_id: "{{ secrets.client_id }}"
+        tenant_id: "{{ tenant_id }}"
+        client_secret: "{{ secrets.client_secret }}"
+        vm_name: "{{ vm_name }}"
+        name: "{{ vm-name }}-image"
+        location: "{{ location }}"
+        state: present
+
+    - name: Delete image
+      azure_rm_image:
+        subscription_id: "{{ subscription_id }}"
+        resource_group_name: "{{ resource_group_name }}"
+        client_id: "{{ secrets.client_id }}"
+        tenant_id: "{{ tenant_id }}"
+        client_secret: "{{ secrets.client_secret }}"
+        name: "{{ vm-name }}-image"
+        location: "{{ location }}"
+        state: absent
+'''
+
+RETURN = '''
```

Comment:
Avoid returning invocation args on the return JSON. Ansible already does this when running with `-vvv` and just duplicates work. I would probably return the id of the new image and other useful information in the return value that is a result of the actions.